src/main/java/uk/gov/companieshouse/api/accounts/interceptor/CompanyAccountInterceptor.java

```diff
@@ -40,34 +50,52 @@
     @SuppressWarnings("unchecked")
     public boolean preHandle(HttpServletRequest request, HttpServletResponse response,
             Object handler) {
+        HttpSession session = request.getSession();
+        Transaction transaction = (Transaction) session
+                .getAttribute(AttributeName.TRANSACTION.getValue());
+        if (transaction == null) {
+            LOGGER.error(
+                    "CompanyAccountInterceptor failed on preHandle: Failed to retrieve a transaction from the session.");
+            return false;
+        }
 
         Map<String, String> pathVariables = (Map) request
                 .getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
         String companyAccountId = pathVariables.get("companyAccountId");
-
         CompanyAccountEntity companyAccountEntity = companyAccountService
                 .findById(companyAccountId);
-        Transaction transaction = (Transaction) request.getSession()
-                .getAttribute(AttributeName.TRANSACTION.getValue());
-        HttpSession session = request.getSession();
+        if (companyAccountEntity == null) {
+            LOGGER.error(
+                    "CompanyAccountInterceptor failed on preHandle: Failed to retrieve a CompanyAccount for "
+                            + transaction
+                            .toString() + ".");
+            return false;
+        }
 
-        if (transaction != null && companyAccountEntity != null) {
-            String accountsLink = transaction.getLinks().get(LinkType.COMPANY_ACCOUNT.getLink());
-            String accountsSelf = companyAccountEntity.getData().getLinks()
-                    .get(LinkType.SELF.getLink());
+        String accountsSelf = companyAccountEntity.getData().getLinks()
+                .get(LinkType.SELF.getLink());
+        Map<String, Resources> resourcesList = transaction.getResources();
+        if (!isLinkInResourceMap(resourcesList, accountsSelf)) {
+            LOGGER.error(
+                    "CompanyAccountInterceptor failed on preHandle: Failed to find the CompanyAccount self link in the Transactions links"
+                            + transaction
+                            .toString() + ".");
```

Comment:
A transaction will grow very big with all of the filings, resources, etc. My opinion is only the necessary data from the transaction should be logged.